templates/cpp/CMakeLists.txt

```diff
@@ -33,209 +33,59 @@ if(XCODE)
-    # whole archive for jni
-    target_link_libraries(${APP_NAME} -Wl,--whole-archive cpp_android_spec -Wl,--no-whole-archive)
-    config_android_shared_libs("dev.axmol.lib" "${CMAKE_CURRENT_SOURCE_DIR}/proj.android/app/src")
-endif()
-
-if (NOT _AX_USE_PREBUILT)
-    target_link_libraries(${APP_NAME} ${_AX_CORE_LIB})
-endif()
-
-# The optional thirdparties(not dependent by engine)
-if (AX_WITH_YAML_CPP)
-    list(APPEND GAME_INC_DIRS "${_AX_ROOT}/3rdparty/yaml-cpp/include")
-endif()
-
-target_include_directories(${APP_NAME} PRIVATE ${GAME_INC_DIRS})
 
+include(AXGameTargetSetup)
 
 # mark app resources, resource will be copy auto after mark
 ax_setup_app_config(${APP_NAME})
 
-if(APPLE)
-    set_target_properties(${APP_NAME} PROPERTIES RESOURCE "${APP_UI_RES}")
-    set_xcode_property(${APP_NAME} INSTALL_PATH "\$(LOCAL_APPS_DIR)")
-    set_xcode_property(${APP_NAME} PRODUCT_BUNDLE_IDENTIFIER "dev.axmol.dummy")
-
-    if(MACOSX)
-        set_target_properties(${APP_NAME} PROPERTIES MACOSX_BUNDLE_INFO_PLIST "${CMAKE_CURRENT_SOURCE_DIR}/proj.ios_mac/mac/Info.plist")
-    elseif(TVOS)
-        set_target_properties(${APP_NAME} PROPERTIES MACOSX_BUNDLE_INFO_PLIST "${CMAKE_CURRENT_SOURCE_DIR}/proj.ios_mac/ios/targets/tvos/Info.plist")
-        set_xcode_property(${APP_NAME} ASSETCATALOG_COMPILER_APPICON_NAME "Brand Assets")
-    elseif(IOS)
-        set_target_properties(${APP_NAME} PROPERTIES MACOSX_BUNDLE_INFO_PLIST "${CMAKE_CURRENT_SOURCE_DIR}/proj.ios_mac/ios/targets/ios/Info.plist")
-        set_xcode_property(${APP_NAME} ASSETCATALOG_COMPILER_APPICON_NAME "AppIcon")
-        set_xcode_property(${APP_NAME} TARGETED_DEVICE_FAMILY "1,2")
-    endif()
-
-    # For code-signing, set the DEVELOPMENT_TEAM:
-    #set_xcode_property(${APP_NAME} DEVELOPMENT_TEAM "GRLXXXX2K9")
-elseif(WINDOWS)
-    if(NOT _AX_USE_PREBUILT)
-        ax_sync_target_dlls(${APP_NAME})
-    endif()
-endif()
+# Add any libraries you need to link to the project after this point
 
-if ((NOT APPLE) AND (NOT WINRT))
-    ax_get_resource_path(APP_RES_DIR ${APP_NAME})
-    ax_sync_target_res(${APP_NAME} LINK_TO ${APP_RES_DIR} FOLDERS ${content_folder} SYM_LINK 1)
-
-    if((WINDOWS AND (NOT (CMAKE_GENERATOR MATCHES "Ninja"))))
-        set_property(TARGET ${APP_NAME} PROPERTY VS_DEBUGGER_WORKING_DIRECTORY "${content_folder}")
-        if(NOT DEFINED BUILD_ENGINE_DONE)
-            set_property(DIRECTORY PROPERTY VS_STARTUP_PROJECT ${APP_NAME})
-        endif()
-    endif()
-elseif(WINRT)
-    if(NOT DEFINED BUILD_ENGINE_DONE)
-        set_property(DIRECTORY PROPERTY VS_STARTUP_PROJECT ${APP_NAME})
-    endif()
-endif()
 
-if (_AX_USE_PREBUILT) # support windows and linux
-    use_ax_compile_define(${APP_NAME})
+# Default Apple-specific setup
+include(AXGameAppleSetup)
 
-    include(AXLinkHelpers)
-    ax_link_cxx_prebuilt(${APP_NAME} ${_AX_ROOT} ${AX_PREBUILT_DIR})
-endif()
+# Default Windows-specific setup
+include(AXGameWindowsSetup)
```

Comment:
could we automatically check platform-spec setup?

Reply:
Do you mean to move the platform checks out of the module and have them like this?

```
if(APPLE)
    # Default Apple-specific setup
    include(AXGameAppleSetup)
elseif(WINDOWS)
    # Default Windows-specific setup
    include(AXGameWindowsSetup)
endif()
```

Reply:
move into , not out, looks like `include(AXGamePlatformSetup)` in project `CMakeLists.txt`

Reply:
Oh, I understand.  The reason I split it was because the Apple section tends to be heavily modified by developers, so I had it in its own file.  I'll merge them together though, since developers can just edit that module directly with their changes.